tests/lookup_registry_test.cpp

```diff
@@ -47,12 +53,18 @@ int main()
 
     // otf2::registry_view<otf2::definition::string, ByCPU> cpu_strings(rego);
 
-    reg.create<otf2::definition::string>(ByCPU(7), std::string("cpu 7"));
-    reg.create<otf2::definition::string>(ByProcess(7435), std::string("process 7435"));
-    reg.create<otf2::definition::string>(ByProcess(745), 17, std::string("process 745"));
+    const auto& str1 = reg.create<otf2::definition::string>(ByCPU(7), std::string("cpu 7"));
+    const auto& str2 =
+        reg.create<otf2::definition::string>(ByProcess(7435), std::string("process 7435"));
+    const auto& str3 =
+        reg.create<otf2::definition::string>(ByProcess(745), 17, std::string("process 745"));
     reg.create<otf2::definition::string>(2, std::string("horst"));
     reg.create<otf2::definition::string>(std::string("regenbogen"));
 
+    reg.create<otf2::definition::system_tree_node>(ByCPU(42), str1, str2);
+    auto& node = reg.get<otf2::definition::system_tree_node>(ByCPU(42));
```

Comment:
```suggestion
    otf2::definition::system_tree_node& node = reg.get<otf2::definition::system_tree_node>(ByCPU(42));
```
Spelling it out here makes the root cause more obvious.